account_tax_one_vat/models/product_template.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _name = "product.template"
+    _inherit = ["product.template", "one.vat.mixin"]
+
+    vat_id = fields.Many2one(
+        "account.tax",
+        compute="_compute_product_vat",
+        string="VAT",
+    )
+    vat = fields.Char(compute="_compute_product_vat", string="VAT name")
+
+    @api.constrains("taxes_id")
+    def _check_only_one_vat_customer_tax(self):
+        self._check_only_one_vat_tax_field("taxes_id")
+
+    @api.constrains("supplier_taxes_id")
+    def _check_only_one_vat_supplier_tax(self):
+        self._check_only_one_vat_tax_field("supplier_taxes_id")
+
```

Comment:
Add onchange when filling the field (like you have on account move line)

```suggestion
    @api.onchange("taxes_id")
    def _onchange_only_one_vat_customer_tax(self):
      self._onchange_one_vat_tax_field("taxes_id")
    
    @api.onchange("supplier_taxes_id")
    def _onchange_only_one_vat_supplier_tax(self):
      self._onchange_one_vat_tax_field("supplier_taxes_id")

_check_only_one_vat_customer_tax
```